pkg/kubelet/dockertools/manager.go

```diff
@@ -1788,7 +1790,14 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, _ api.PodStatus, podStatus *kubec
 
 		// Find the pod IP after starting the infra container in order to expose
 		// it safely via the downward API without a race and be able to use podIP in kubelet-managed /etc/hosts file.
-		pod.Status.PodIP = dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		podIP := dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		if podIP == "" {
+			glog.Errorf("Pod IP for %v/%v could not be determined", pod.Namespace, pod.Name)
```

Comment:
This will fire for net=host containers until https://github.com/kubernetes/kubernetes/pull/20578 gets merged, since docker reports no address for these containers.  Perhaps just do the Infof().